src/Raven.Server/Documents/Handlers/ShardedDocumentHandler.cs

```diff
@@ -1,20 +1,30 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO.Compression;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Text;
 using System.Threading.Tasks;
 using Raven.Client;
+using Raven.Client.Documents.Commands;
 using Raven.Client.Documents.Commands.Batches;
+using Raven.Server.Documents.Sharding;
 using Raven.Server.Extensions;
 using Raven.Server.Routing;
 using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
 
 namespace Raven.Server.Documents.Handlers
 {
     public class ShardedDocumentHandler : ShardedRequestHandler
     {
+        protected readonly StringBuilder UrlStringBuilder = new StringBuilder();
+        protected readonly List<Task> Task = new List<Task>();
+        protected readonly Dictionary<int, ShardedCommand> Cmds = new Dictionary<int, ShardedCommand>();
+        protected readonly List<TransactionOperationContext> CmdsContext = new List<TransactionOperationContext>();
```

Comment:
Why do we want these to be instance fields?

Reply:
The handlers are long-living aren't they?
I want to "object-pool " those since they are managed objects are going to be generated a lot

Reply:
They aren't long lived. They are for a single request.
And if they were, you would have thread safety issues